src/Controller/LoginController.php

```diff
@@ -0,0 +1,34 @@
+<?php declare(strict_types=1);
+
+namespace Pdsinterop\Solid\Controller;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+
+class LoginController extends AbstractController
+{
+    final public function __invoke(ServerRequestInterface $request, array $args): ResponseInterface
+    {
+        $postBody = $request->getParsedBody();
+        $response = $this->getResponse();
+        // var_dump($_SESSION);
+        if (isset($_SESSION["userid"])) {
```

Comment:
Not per-se for now (can also be fixed lateR), but unless the string needs to be parsed by PHP, the current convention is to use single quotes `'`:

```php
         if (isset($_SESSION['userid'])) { 
```

This goes for the strings below as well, except `"<h1>Welcome $user</h1>\n"`, as the double quote `"` is needed to make the linebreak work.